### PR TITLE
Add check to exit training on NaN

### PR DESCRIPTION
**Pull request by gkielian**
This will speed up curriculums and allow for higher-risk, higher-reward model settings (higher learning rate, higher weight decay, post-norm, etc).

Also fixes issue: https://github.com/ReaLLMASIC/nanoGPT/issues/92

### CODE REVIEW

**Review by klei22 — APPROVED** [review]

**Comment by klei22** [comment]
Looks good
